### PR TITLE
BZ-1851054 Fixed CLI secondary network attachment procedure 

### PR DESCRIPTION
**Pull request by sjhala-ccs**
BZ-1851054  https://bugzilla.redhat.com/show_bug.cgi?id=1851054

Preview: https://bz-1851054--ocpdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/vm_networking/virt-attaching-vm-multiple-networks.html#virt-creating-bridge-nad-cli_virt-attaching-multiple-networks

Fixed CLI procedure to connect VM to secondary network

Tagged Meni Yakove for QE review

Applies to 4.5 docs

### CODE REVIEW

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 1]
```diff
@@ -25,53 +25,69 @@ apiVersion: "k8s.cni.cncf.io/v1"
 kind: NetworkAttachmentDefinition
 metadata:
   name: a-bridge-network
-  annotations:
-    k8s.v1.cni.cncf.io/resourceName: bridge.network.kubevirt.io/br0 <1>
 spec:
   config: '{
     "cniVersion": "0.3.1",
-    "name": "cnv-bridge-conf", <2>
+    "name": "cnv-bridge-conf", <1>
```
Comment (aburdenthehand) [inline modules/virt-creating-bridge-nad-cli.adoc:31]: Not sure if it's strictly necessary but this should probably match the name of the NAD (a-bridge-network)
@phoracek will keep me honest.

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 2]
```diff
@@ -25,53 +25,69 @@ apiVersion: "k8s.cni.cncf.io/v1"
...
-  name: example-vm
-  annotations:
-    k8s.v1.cni.cncf.io/networks: a-bridge-network <1>
-spec:
-...
+interfaces:
+- masquerade: {}
+  name: default
+- bridge: {}
+  name: bridge-net
```
Comment (aburdenthehand) [inline modules/virt-creating-bridge-nad-cli.adoc:70]: This should be replaceable (wrapped in <>) and correspond to the name used above (a-bridge-network)

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 3]
```diff
@@ -25,53 +25,69 @@ apiVersion: "k8s.cni.cncf.io/v1"
...
 you want to use a different CNI.
-<4> You must substitute the actual name of the bridge, if it is not `br0`.
-<5> Required. This allows the MAC pool manager to assign a unique MAC address to the connection.
+<3> You must substitute the actual name of the bridge, if it is not `br0`.
+<4> Required. This allows the MAC pool manager to assign a unique MAC address to the connection.
++
+[NOTE]
+====
+The virtual machine instance will be attached to the bridge `br0` through an
+access port with the requested VLAN.
```
Comment (phoracek) [inline modules/virt-creating-bridge-nad-cli.adoc:53]: we don't request any VLAN in the example

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 4]
```diff
@@ -25,53 +25,69 @@ apiVersion: "k8s.cni.cncf.io/v1"
...
 +
 ----
 $ oc create -f <resource_spec.yaml>
 ----
 
-. Edit the configuration file of a virtual machine or virtual machine instance that you want to connect to the
-bridge network:
+. Edit the virtual machine instance configuration file to include the details of the interface and network.
+
+.. Specify the network and MAC address, if required by the server. If the MAC address is not specified, a value is assigned
```
Comment (phoracek) [inline modules/virt-creating-bridge-nad-cli.adoc:62]: This is conditioned by the namespace being opted in to MAC pool.

Let's just drop the MAC notion, at least from this PR. Usually, user would not need to set it.

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 5]
```diff
@@ -25,53 +25,69 @@ apiVersion: "k8s.cni.cncf.io/v1"
...
-apiVersion: v1
-kind: VirtualMachine
-metadata:
-  name: example-vm
-  annotations:
-    k8s.v1.cni.cncf.io/networks: a-bridge-network <1>
-spec:
-...
+interfaces:
+- masquerade: {}
```
Comment (phoracek) [inline modules/virt-creating-bridge-nad-cli.adoc:67]: Worth a note that this VM will be connected to both the default pod network and to the bridge.

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 6]
```diff
@@ -25,53 +25,69 @@ apiVersion: "k8s.cni.cncf.io/v1"
...
+  name: default
+- bridge: {}
+  name: bridge-net
+  macAddress: de:00:00:00:00:de
+  bootOrder: 1
```
Comment (phoracek) [inline modules/virt-creating-bridge-nad-cli.adoc:72]: Not interesting outside of the PXE context.

**Review by phoracek — CHANGES_REQUESTED** [review]

**Comment by sjhala-ccs** [comment]
@phoracek I have incorporated your comments. Can you please review again? Thanks.

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 7]
```diff
@@ -44,33 +44,36 @@ spec:
...
 ----
-apiVersion: v1
-kind: VirtualMachine
-metadata:
-  name: example-vm
-  annotations:
-    k8s.v1.cni.cncf.io/networks: a-bridge-network <1>
-spec:
-...
+networks:
```
Comment (phoracek) [inline modules/virt-creating-bridge-nad-cli.adoc:62]: We need to keep the YAML example with `interfaces` list. Both `networks` and `interfaces` is needed to build the configuration.

**Review by phoracek — CHANGES_REQUESTED** [review]

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 8]
```diff
@@ -44,33 +44,51 @@ spec:
...
+    - masquerade: {}
+      name: default
+    - bridge: {}
+      name: bridge-net
+...
+
+  networks:
+    - name: default
+      pod: {}
+    - name: bridge-net
```
Comment (phoracek) [inline modules/virt-creating-bridge-nad-cli.adoc:79]: Would be worth marking this with a note and connecting it to the same name on line 73.

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 9]
```diff
@@ -44,33 +44,51 @@ spec:
...
 apiVersion: v1
 kind: VirtualMachine
 metadata:
-  name: example-vm
-  annotations:
-    k8s.v1.cni.cncf.io/networks: a-bridge-network <1>
+    name: example-vm
 spec:
+...
+  interfaces:
```
Comment (phoracek) [inline modules/virt-creating-bridge-nad-cli.adoc:69]: `interfaces` are not directly under `spec`. They are under `spec.domain.devices`.

**Review by phoracek — CHANGES_REQUESTED** [review]

**Review by phoracek — APPROVED** [review]
/lgtm

Thanks

**Comment by sjhala-ccs** [comment]
@myakove Can you please review? Thanks.

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 10]
```diff
@@ -44,33 +44,53 @@ spec:
...
 ...
 ----
-<1> You must substitute the actual `name` value from the
+<1> The `name` value for the bridge interface and network must be the same.
+<2> You must substitute the actual `name` value from the
 NetworkAttachmentDefinition.
++
+[NOTE]
+====
+The virtual machine instance will be connected to both the `default` Pod network and `bridge-net` which is
```
Comment (ousleyp) [inline modules/virt-creating-bridge-nad-cli.adoc:91]: This note is kind of hard to parse -- could it be split into two sentences?

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 11]
```diff
@@ -44,33 +44,53 @@ spec:
...
 <5> Required. This allows the MAC pool manager to assign a unique MAC address to the connection.
+
 +
 ----
 $ oc create -f <resource_spec.yaml>
 ----
 
-. Edit the configuration file of a virtual machine or virtual machine instance that you want to connect to the
-bridge network:
+. Edit the configuration file of a virtual machine or virtual machine instance that you want to connect to the bridge network.
```
Comment (ousleyp) [inline modules/virt-creating-bridge-nad-cli.adoc:59]: Maybe say "For example:" at the end of this step?

**Review by ousleyp — COMMENTED** [review]
Just a couple of nits, otherwise LGTM.
